src/workers/notifications/postDeleted.ts

```diff
@@ -0,0 +1,47 @@
+import { messageToJson, Worker } from '../worker';
+import { Notification, Post } from '../../entity';
+import { ChangeObject } from '../../types';
+
+interface Data {
+  post: ChangeObject<Post>;
+}
+
+const worker: Worker = {
+  subscription: 'api.post-deleted-notification-cleanup',
+  handler: async (message, con, logger) => {
+    const data: Data = messageToJson(message);
+    const { post } = data;
+
+    const databasePost = await con
+      .getRepository(Post)
+      .findOneBy({ id: post?.id });
+    if (!databasePost || !databasePost?.deleted) {
+      return logger.info(
+        {
+          data,
+          messageId: message.messageId,
+        },
+        'failed to delete notifications due to post deletion',
+      );
+    }
+
+    await con
+      .getRepository(Notification)
+      .createQueryBuilder()
+      .delete()
+      .where({
+        referenceType: 'post',
```

Comment:
What about comments from the post such as replies and mentions. Should we delete those as well when the post gets deleted?

Reply:
I think that's already handled by the system afaik.

Reply:
My bad, I was not clear. I mean the notifications from those comments.

Reply:
Oh, wait. You are right. There will be a full cycle. Post deleted -> cascades to delete comments -> worker triggered -> delete notification comments. Great work!

Reply:
Yes should cascade the comment deletion right? 🙏
Thanks for double checking though

Reply:
Deleting posts is just a logical deletion, no cascade will happen in this case. We need to delete the relevant comment notifications manually. That's a great catch Lee! We need a follow-up PR I guess

Reply:
Oh, right. I forgot we don't actually hard-delete post data 😅

Reply:
Last time we talked to product they asked to keep the comments on deletion but we can double check